src/pip/_internal/commands/install.py

```diff
@@ -732,3 +737,25 @@ def create_os_error_message(error, show_traceback, using_user_site):
         parts.append(".\n")
 
     return "".join(parts).strip() + "\n"
+
+
+def prepare_sudo_warning():
+    # type: () -> None
+    """Output a warning for sudo users on Unix.
+
+    On Windows, users may run pip as Administrator without issues.
+    """
+    try:
```

Comment:
Why is this wrapped in a try?

Reply:
See next comment for how I'd suggest rewriting this function.

Reply:
The try is for `os.getuid()`, which throws an AttributeError when run on Windows. Replied in your other comment.